BlueprintUI/Sources/Internal/SPL/SPCacheTree.swift

```diff
@@ -55,38 +55,136 @@ final class SPCacheTree<Key, Value, SubcacheKey> where Key: Hashable, SubcacheKe
+        self.options = options
     }
 }
 
-extension SPValueCache where Key == SizeConstraint {
+struct SPCacheOptions {
+    var hintRangeBoundaries: Bool
+    var searchUnconstrainedKeys: Bool
+}
+
+extension SPValueCache where Key == SizeConstraint, Value == CGSize {
 
     func get(key: Key, or create: (Key) -> Value) -> Value {
         if let size = values[key] {
             Logger.logCacheHit(object: self, description: path, constraint: key)
             return size
         }
         
+        // TODO:
+        // This has a negative perf impact on deep stack tests. Need to determine what the impact
+        // is on a wider variety of tests, and if there is a way we can avoid this for cases
+        // where it doesn't help (e.g. not in a scroll view).
+        if options.searchUnconstrainedKeys {
+            if
+                case .atMost(let maxHeight) = key.height,
+                let size = values[.init(width: key.width, height: .unconstrained)],
+                size.height <= maxHeight
+            {
+                Logger.logCacheUnconstrainedSearchMatch(
+                    object: self,
+                    description: path,
+                    constraint: key,
+                    match: .init(width: key.width, height: .unconstrained)
+                )
+                values[key] = size
+                return size
+            }
+            
+            if
+                case .atMost(let maxWidth) = key.width,
+                let size = values[.init(width: .unconstrained, height: key.height)],
+                size.width <= maxWidth
+            {
+                Logger.logCacheUnconstrainedSearchMatch(
+                    object: self,
+                    description: path,
+                    constraint: key,
+                    match: .init(width: .unconstrained, height: key.height)
+                )
+                // TODO: test with and without caching this
+                values[key] = size
+                return size
+            }
+            
+            if
+                case .atMost(let maxWidth) = key.width,
+                case .atMost(let maxHeight) = key.height,
+                let size = values[.unconstrained],
+                size.width <= maxWidth,
+                size.height <= maxHeight
+            {
+                Logger.logCacheUnconstrainedSearchMatch(
+                    object: self,
+                    description: path,
+                    constraint: key,
+                    match: .unconstrained
+                )
+                values[key] = size
+                return size
+            }
+        }
+        
         Logger.logCacheMiss(object: self, description: path, constraint: key)
         
         Logger.logMeasureStart(object: self, description: path, constraint: key)
         let size = create(key)
         Logger.logMeasureEnd(object: self)
 
         values[key] = size
+        
+        if options.hintRangeBoundaries {
```

Comment:
Do we anticipate a use case for disabling this? Are we thinking about feature flagging this?

Reply:
The case I had in mind was to be able to run performance tests with it turned on & off.